node/external/transactionAPI/unmarshaller.go

```diff
@@ -92,6 +96,7 @@ func (tu *txUnmarshaller) unmarshalTransaction(txBytes []byte, txType transactio
 	apiTx.Receivers = datafield.EncodeBytesSlice(tu.addressPubKeyConverter.Encode, res.Receivers)
 	apiTx.ReceiversShardIDs = res.ReceiversShardID
 	apiTx.IsRelayed = res.IsRelayed
+	apiTx.ChainID = tu.chainID
```

Comment:
this should be on the initial transaction.

Reply:
Refactored